src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala

```diff
@@ -275,20 +277,25 @@ class JavaUrlConnectionRequestForwarder @Inject() (
       } else {
         val leaderConnection: HttpURLConnection = createAndConfigureConnection(url)
         try {
-          copyRequestToConnection(leaderConnection, request)
-          copyConnectionResponse(
-            response
-          )(
+          copyRequestToConnection(leaderConnection, request) match {
+            case Failure(ex: ConnectException) =>
+              log.error(ERROR_STATUS_CONNECTION_REFUSED, ex)
+              response.sendError(BadGateway.intValue, ERROR_STATUS_CONNECTION_REFUSED)
+              return
```

Comment:
Can we rework this code to remove the need for this return statement? The multiple finally statements, plus the return, seem to make the control of this method pretty difficult to follow.